adc/ad7779/imxrt/spi.c

```diff
@@ -0,0 +1,60 @@
+#include <unistd.h>
+#include <sys/msg.h>
+
+#include <phoenix/arch/imxrt.h>
+#include <imxrt-multi.h>
+
+#include "../ad7779.h"
+
+static oid_t spi_driver;
+
+
+int spi_exchange(uint8_t *buff, uint8_t len)
+{
+	msg_t msg = { 0 };
+	multi_i_t *imsg = (multi_i_t *)msg.i.raw;
+	multi_o_t *omsg = (multi_o_t *)msg.o.raw;
+
+	msg.type = mtDevCtl;
+	msg.i.data = buff;
+	msg.i.size = len;
+	msg.o.data = buff;
+	msg.o.size = len;
+
+	imsg->id = id_spi4;
+	imsg->spi.type = spi_transaction;
+	imsg->spi.transaction.cs = 0;
+	imsg->spi.transaction.frameSize = len;
+
+	msgSend(spi_driver.port, &msg);
+
+	return omsg->err;
+}
+
+
+int spi_init(void)
+{
+	msg_t msg;
+	multi_i_t *imsg = (multi_i_t *)msg.i.raw;
+
+	if (spi_driver.port == 0) {
+		while (lookup("/dev/gpio1", NULL, &spi_driver) < 0)
```

Comment:
For convenience SPI devices should be accessed via `/dev/spiN`, in Your case `N=4`. I know this may work because of common [(dispach message handler)](https://github.com/phoenix-rtos/phoenix-rtos-devices/blob/64e1fa95b44487b9e9850517d9104992be562ba3/multi/imxrt-multi/imxrt-multi.c#L98) for gpio, spi and i2c, but if e.g. [`SPI4`](https://github.com/phoenix-rtos/phoenix-rtos-devices/blob/64e1fa95b44487b9e9850517d9104992be562ba3/multi/imxrt-multi/imxrt-multi.c#L274-L277) is disabled in `BOARD_CONFIG` the function should fail or give up after counting down retries, and not use kind of backdoor through `/dev/gpio1` (which seems to be[ always ](https://github.com/phoenix-rtos/phoenix-rtos-devices/blob/64e1fa95b44487b9e9850517d9104992be562ba3/multi/imxrt-multi/imxrt-multi.c#L250-L255)registered with `imxrt-multi`), while trying to use `/dev/spi4`.